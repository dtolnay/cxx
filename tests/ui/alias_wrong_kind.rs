#[cxx::bridge]
mod here {
    struct Shared {
        z: usize,
    }

    extern "C" {
        type C;
    }
}

// Rustfmt mangles the extern type alias.
// https://github.com/rust-lang/rustfmt/issues/4159
#[rustfmt::skip]
#[cxx::bridge]
mod there {
    type C = crate::here::C;

    extern "C" {
        type Shared = crate::here::Shared;
    }
}

fn main() {}
