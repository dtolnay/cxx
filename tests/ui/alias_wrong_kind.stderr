error[E0271]: type mismatch resolving `<here::C as ExternType>::Kind == KindShared`
   --> $DIR/alias_wrong_kind.rs:15:1
    |
15  | #[cxx::bridge]
    | ^^^^^^^^^^^^^^ expected struct `KindShared`, found struct `KindOpaqueCpp`
    |
   ::: $WORKSPACE/src/extern_type.rs
    |
    | pub fn verify_extern_type<T: ExternType<Kind = Kind, Id = Id>, Kind, Id>() {}
    |                                         ----------- required by this bound in `verify_extern_type`

error[E0271]: type mismatch resolving `<here::Shared as ExternType>::Kind == KindOpaqueCpp`
   --> $DIR/alias_wrong_kind.rs:15:1
    |
15  | #[cxx::bridge]
    | ^^^^^^^^^^^^^^ expected struct `KindOpaqueCpp`, found struct `KindShared`
    |
   ::: $WORKSPACE/src/extern_type.rs
    |
    | pub fn verify_extern_type<T: ExternType<Kind = Kind, Id = Id>, Kind, Id>() {}
    |                                         ----------- required by this bound in `verify_extern_type`
